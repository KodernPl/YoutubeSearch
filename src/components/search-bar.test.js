import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import SearchBar from './search-bar';

const onSearchTermChangeMock = jest.fn();

beforeEach(() => {
  onSearchTermChangeMock.mockReset();
});

const renderComponent = () => {
  return mount((
    <SearchBar
      onSearchTermChange={onSearchTermChangeMock}
    />
  ));
};

test('renders correctly', () => {
  const component = renderComponent();

  expect(toJson(component)).toMatchSnapshot();
});

test('should call onSearchTermChange when text is entered', () => {
  const component = renderComponent();
  component.find('input').simulate('change', { target: { value: 'TEST' } });

  expect(onSearchTermChangeMock).toBeCalledWith('TEST');
});
