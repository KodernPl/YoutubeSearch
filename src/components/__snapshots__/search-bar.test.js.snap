// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<SearchBar
  onSearchTermChange={[MockFunction]}
>
  <search-bar__SearchBarWrapper>
    <div
      className="search-bar__SearchBarWrapper-sc-1op3id8-0 fbColO"
    >
      <search-bar__Input
        onChange={[Function]}
        placeholder="Enter Search"
        value=""
      >
        <input
          className="search-bar__Input-sc-1op3id8-1 ksdLow"
          onChange={[Function]}
          placeholder="Enter Search"
          value=""
        />
      </search-bar__Input>
    </div>
  </search-bar__SearchBarWrapper>
</SearchBar>
`;
